hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestFailureHandlingByClient.java

```diff
@@ -163,6 +165,42 @@ public void shutdown() {
     }
   }
 
+  @Test
+  public void testDNExcludeListExpire() throws Exception {
+    startCluster();
+    String keyName = "test_key";
+    OzoneOutputStream key = createKey(keyName, ReplicationType.RATIS, 0);
+
+    byte[] data = "test_key_data".getBytes(UTF_8);
+    key.write(data);
+
+    KeyOutputStream groupOutputStream =
+        (KeyOutputStream) key.getOutputStream();
+    List<OmKeyLocationInfo> locationInfoList =
+        groupOutputStream.getLocationInfoList();
+
+    long containerId = locationInfoList.get(0).getContainerID();
+    ContainerInfo container = cluster.getStorageContainerManager()
+        .getContainerManager()
+        .getContainer(ContainerID.valueOf(containerId));
+    Pipeline pipeline =
+        cluster.getStorageContainerManager().getPipelineManager()
+            .getPipeline(container.getPipelineID());
+    List<DatanodeDetails> datanodes = pipeline.getNodes();
+    cluster.shutdownHddsDatanode(datanodes.get(0));
```

Comment:
1. We can bring down DNs before key.write(data)
2. perform write data (in Async way), it will fail and retry keep happen and verify if exclude list populated
3. restart DN
4. After some time, exclude list should be cleared and write gets success.

StaleNode/Deadnode: should be 30sec
exclude list timeout: 5 sec

We do not need CustomClock in this case...